3ds/libdl/dlfcn.c

```diff
@@ -0,0 +1,36 @@
+#include "dlfcn.h"
+
+#include <stddef.h>
+
+static dl_open open = NULL;
```

Comment:
Nitpick, but `= NULL` does nothing in this context and can be left out.
Also, you might want to rename these to something like `g_open` that do not have generic names that conflict with existing well known POSIX symbols; and yes, I'm aware that these are static vars, however it can be confusing to read.